cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/test/java/com/sap/cloud/cf/monitoring/java/CustomMetricRegistryTest.java

```diff
@@ -0,0 +1,95 @@
+import com.sap.cloud.cf.monitoring.client.configuration.CFConfigurationProvider;
+import com.sap.cloud.cf.monitoring.client.configuration.EnvUtils;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CustomMetricRegistryTest {
+    private static final String TEST_INSTANCE_GUID = UUID.randomUUID().toString();
+    private static final String TEST_APPLICATION_GUID = UUID.randomUUID().toString();
+    private static final String TEST_INSTANCE_INDEX = "0";
+    private static final String TEST_MONITORING_URL = "https://localhost";
+    private static final String TEST_CLIENT_ID = UUID.randomUUID().toString();
+    private static final String TEST_SECRET = "super-secure-secret";
+
+    private static final String[] CF_INSTANCE_GUID_KEY_ENV =
+        { CFConfigurationProvider.CF_INSTANCE_GUID_KEY, TEST_INSTANCE_GUID };
+    private static final String[] CF_INSTANCE_INDEX_KEY_ENV =
+        { CFConfigurationProvider.CF_INSTANCE_INDEX_KEY, TEST_INSTANCE_INDEX };
+    private static final String[] VCAP_APPLICATION_KEY_ENV = { CFConfigurationProvider.VCAP_APPLICATION_KEY,
+            "{\"application_id\":\"" + TEST_APPLICATION_GUID + "\",\"application_name\":\"monitoring-sample-app\"}" };
+    private static final String[] VCAP_SERVICES_KEY_ENV = { CFConfigurationProvider.VCAP_SERVICES_KEY,
+            "{\"application-logs\": [{\"credentials\": { \"" + METRICS_COLLECTOR_CLIENT_ID_KEY + "\" : \""
+                    + TEST_CLIENT_ID + "\", \"" + METRICS_COLLECTOR_CLIENT_SECRET_KEY + "\": \"" + TEST_SECRET
+                    + "\", \"" + METRICS_COLLECTOR_URL_KEY + "\": \"" + TEST_MONITORING_URL + "\"}}]}" };
+
+    @Before
+    public void setUp() throws Exception {
+        Field inst = CustomMetricRegistry.class.getDeclaredField("instance");
+        inst.setAccessible(true);
+        inst.set(null, null);
+    }
+
+    @Test
+    public void testCustomMetricRegistryInitializeWithEnvs() throws Exception {
+        EnvUtils.setEnvs(new String[][] { CF_INSTANCE_GUID_KEY_ENV, CF_INSTANCE_INDEX_KEY_ENV, VCAP_SERVICES_KEY_ENV,
+                VCAP_APPLICATION_KEY_ENV });
+        CustomMetricRegistry metricRegistry = (CustomMetricRegistry) CustomMetricRegistry.get();
+
+        assertNotNull(metricRegistry.getReporter());
+    }
+
+    @Test
+    public void testCustomMetricRegistryInitializeWithoutEnvs() throws Exception {
+        EnvUtils.setEnvs(new String[][] {});
+        MetricRegistry metricRegistry = CustomMetricRegistry.get();
+        MetricRegistry metricRegistry1 = CustomMetricRegistry.get();
+
+        assertTrue(metricRegistry instanceof CustomMetricRegistry);
+        assertEquals(metricRegistry, metricRegistry1);
+        assertNull(((CustomMetricRegistry) metricRegistry).getReporter());
+    }
+
+    @Test
+    public void testCustomMetricRegistryInitializeWithEnvsAndDisabledFlag() throws Exception {
+        EnvUtils.setEnvs(new String[][] { CF_INSTANCE_GUID_KEY_ENV, CF_INSTANCE_INDEX_KEY_ENV, VCAP_SERVICES_KEY_ENV,

```

Comment:
Here is an example, where the key-value pairs used in the test are completely obfuscated by the chosen signature of `String[][]`.